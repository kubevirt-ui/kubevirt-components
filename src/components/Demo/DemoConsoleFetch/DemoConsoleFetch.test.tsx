import * as React from 'react';

import { cleanup, render, screen } from '@testing-library/react';

import { DemoConsoleFetch } from './DemoConsoleFetch';

afterEach(cleanup);

const name = 'vm-example';
const namespace = 'default';
const errorText = 'Missing virtual machine data';
const dataTestID = 'DemoUseResourceTest';

test('Render DemoConsoleFetch (valid vm)', async () => {
  const { asFragment } = render(
    <DemoConsoleFetch
      name={name}
      namespace={namespace}
      errorText={errorText}
      dataTestID={dataTestID}
    />,
  );

  await screen.findByTestId(dataTestID);
  expect(asFragment()).toMatchSnapshot();
});

test('Render DemoConsoleFetch (none valid vm)', async () => {
  const { asFragment } = render(
    <DemoConsoleFetch
      name="does-not-exist"
      namespace="default"
      errorText={errorText}
      dataTestID={dataTestID}
    />,
  );

  await screen.findByTestId(`${dataTestID}-bad-response`);
  expect(asFragment()).toMatchSnapshot();
});

test('Render DemoConsoleFetch (missing vm)', async () => {
  const { asFragment } = render(
    <DemoConsoleFetch
      name="does-not-exist"
      namespace="does-not-exist"
      errorText={errorText}
      dataTestID={dataTestID}
    />,
  );

  await screen.findByTestId(`${dataTestID}-error`);
  expect(asFragment()).toMatchSnapshot();
});
