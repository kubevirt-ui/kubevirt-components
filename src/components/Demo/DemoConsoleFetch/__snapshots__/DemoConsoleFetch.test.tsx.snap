// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render DemoConsoleFetch (missing vm) 1`] = `
<DocumentFragment>
  <small
    class=""
    data-ouia-component-id="OUIA-Generated-Text-3"
    data-ouia-component-type="PF4/Text"
    data-ouia-safe="true"
    data-pf-content="true"
    data-test-id="DemoUseResourceTest-error"
  >
    Missing virtual machine data
  </small>
</DocumentFragment>
`;

exports[`Render DemoConsoleFetch (none valid vm) 1`] = `
<DocumentFragment>
  <small
    class=""
    data-ouia-component-id="OUIA-Generated-Text-2"
    data-ouia-component-type="PF4/Text"
    data-ouia-safe="true"
    data-pf-content="true"
    data-test-id="DemoUseResourceTest-bad-response"
  >
    Missing virtual machine data
  </small>
</DocumentFragment>
`;

exports[`Render DemoConsoleFetch (valid vm) 1`] = `
<DocumentFragment>
  <span
    data-test-id="DemoUseResourceTest"
  >
    <div
      class="pf-c-label-group"
    >
      <div
        class="pf-c-label-group__main"
      >
        <ul
          aria-label="Label group category"
          class="pf-c-label-group__list"
          role="list"
        >
          <li
            class="pf-c-label-group__list-item"
          >
            <span
              class="pf-c-label"
            >
              <a
                class="pf-c-label__content"
                href="#"
              >
                <span
                  class="pf-c-label__text"
                >
                  VMINotExists=False
                </span>
              </a>
            </span>
          </li>
        </ul>
      </div>
    </div>
  </span>
</DocumentFragment>
`;
